diff --git a/hw/mcu/stm/stm32l4xx/src/hal_flash.c b/hw/mcu/stm/stm32l4xx/src/hal_flash.c
index 7330d5974..b263c20dd 100644
--- a/hw/mcu/stm/stm32l4xx/src/hal_flash.c
+++ b/hw/mcu/stm/stm32l4xx/src/hal_flash.c
@@ -38,9 +38,12 @@ stm32_mcu_flash_erase_sector(const struct hal_flash *dev, uint32_t sector_addres
         eraseinit.TypeErase = FLASH_TYPEERASE_PAGES;
         if ((sector_address - dev->hf_base_addr) < (_FLASH_SIZE / 2)) {
             eraseinit.Banks = FLASH_BANK_1;
-        } else {
+        }
+#ifdef FLASH_BANK_2
+        else {
             eraseinit.Banks = FLASH_BANK_2;
         }
+#endif
         eraseinit.Page = (sector_address - dev->hf_base_addr) / FLASH_PAGE_SIZE;
         eraseinit.NbPages = 1;
         rc = HAL_FLASHEx_Erase(&eraseinit, &PageError);
